plugin/codeclimate.vim

```diff
@@ -30,17 +30,21 @@ function! s:AnalyzeOpenFiles()
     call s:ModifiedFilesWarning()
   endif
 
+  let l:extraopts = s:BufferAnalysisOpts(l:bufs)
   let l:files = s:BufferNames(l:bufs)
+  let l:cli = join(l:extraopts, ' ') . ' ' . join(map(l:files, 'shellescape(v:val)'), ' ')
 
-  call s:RunAnalysis(join(map(l:files, 'shellescape(v:val)'), ' '))
+  call s:RunAnalysis(l:cli)
 endfunction
 
 function! s:AnalyzeCurrentFile()
   if &modified
     call s:ModifiedFilesWarning()
   endif
 
-  call s:RunAnalysis(shellescape(@%))
+  let l:cli = getbufvar(@%, 'codeclimateflags') . ' ' . shellescape(@%)
```

Comment:
Is it possible to use the `BufferAnalysisOpts` function below, passing in only the current file instead of duplicating some of this `getbufvar` logic?